### ISSUE TITLE
BSOD when trying to access Options

### ISSUE THREAD

**Issue by Windrider6** [Max-Github/FireFoxInvertColors#37]
Windows 10 Pro, 64 bit
Firefox version: 83.0 (64-bit)
Invert Colors version: 0.3.0.0
BSOD log and MiniDump: [2020-11-18 BSOD - Invert Colors addon in Firefox.zip](https://github.com/Max-Github/FireFoxInvertColors/files/5563491/2020-11-18.BSOD.-.Invert.Colors.addon.in.Firefox.zip)

BSOD happened when I tried to go to the options for "Invert Colors" addon in Firefox.  Area where options are supposed to show was black (using Windows & Firefox dark theme), and windows scroll showed that the area was getting bigger.

After reboot, tried the same thing again, and this time the computer froze up with no output to monitor.  No BSOD.  Had to force power-off of computer by pressing power button on it for more than 3 seconds.

**Comment by Max-Github** [comment]
That is a very strange occurrence. Did not even think that this would be possible with this simple addon.

This might need to be forwarded to the Firefox team.

**Comment by Windrider6** [comment]
It doesn't happen with any other addon.

**Comment by Hellosager** [comment]
Hi, I looked at your files and saw the error message: "The computer has rebooted from a bugcheck.  The bugcheck was: 0x000000ea (0xffffe408eb82a080, 0x0000000000000000, 0x0000000000000000, 0x0000000000000000)"

Google told me that many other people had this problem in the past (only found things around 2016-2018) if they use AMD Radeon drivers. Since you also use Radeon drivers this could be related.

If you google the FAILURE_ID_HASH ea458ad2-d5ab-aa6c-7a11-54653c70dfb8 that I found in the dmp-file you get even more results indicating this is an issue with AMD drivers. The error itself states this problem is most likely hardware or driver related.

A device driver is spinning in an infinite loop, most likely waiting for hardware to become idle.

"This usually indicates problem with the hardware itself, or with the device driver programming the hardware incorrectly. Frequently, this is the result of a bad video card or a bad display driver."
(https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-check-0xea--thread-stuck-in-device-driver)

So I dont think this is directly related to the addon or to firefox (even thought it seems weird this only happens to the addon settings). What I recommend you to do is to reinstall the driver for your AMD graphics card. I found other possible solutions (for exaample https://community.amd.com/t5/graphics/0x000000ea-error-after-updating-drivers-win-10-64/td-p/211201) but this seems to be the most straight forward for me at first.

TL;DR: reinstall AMD drivers

**Comment by Windrider6** [comment]
Thank you for your response.  Still problem.

A couple days before I discovered this problem, I had done a clean reinstall of Windows 10, installing the most recent drivers from the AMD website.  Now, I let Windows Update install a couple of AMD drivers it wanted to, rebooted, then had Windows update the video driver.  That driver is also labeled as AMD, but two years newer (2019).

Now when trying to view Invert Colors options, the computer freezes up and I have to force a power off.  I noticed that the freeze-up doesn't happen immediately, so I was able to use a screen recorder to record what happens, stopping the recording just before the freeze-up.  MKV file zipped: [Invert Colors.zip](https://github.com/Max-Github/FireFoxInvertColors/files/5599126/Invert.Colors.zip)

I noticed that before the options area goes black, there is a flash of "File not Found", but I doubt that has anything to do with the issue, since it goes away quick and it looks like the window is loading something.

At this point, I'm going to give up and not try to see or use those options.  It is easier for my mental health than trying to test more video drivers.

**Comment by Hellosager** [comment]
Hi @Windrider6,
I just noticed the options you would see in the browser tab were moved into a popup with the commit [792f7bb5120e886a98b8b1c081d4ba67ed49c8c3](https://github.com/Max-Github/FireFoxInvertColors/commit/792f7bb5120e886a98b8b1c081d4ba67ed49c8c3).

So while this doesn't explain your Windows crashs it explains the "File not found" (because the options file was removed and so the browser can't find it) and it enables you to use the options of the addon by clicking it's icon in the top right of firefox:

![grafik](https://user-images.githubusercontent.com/21982349/134017993-ff3a6546-2e2f-4501-b3e9-683268580d87.png)

Maybe you recognized that by yourself already or you stopped using the addon. Just to let you know in case you still need it.

Have a nice day

**Comment by Windrider6** [comment]
I use the add-on "Dark Reader" now, which works wonderfully.  Better than this add-on.
